web/packages/shared/components/TerminalSearch/TerminalSearch.tsx

```diff
@@ -19,90 +19,103 @@
 import { useEffect, useState, useRef, useCallback } from 'react';
 import styled, { useTheme } from 'styled-components';
 import { Flex, ButtonIcon, Box, P2, Input } from 'design';
-import { Terminal as Xterm } from '@xterm/xterm';
-import { ISearchOptions } from '@xterm/addon-search';
 import { ArrowDown, ArrowUp, Cross } from 'design/Icon';
+import { SearchAddon } from '@xterm/addon-search';
+
+export interface TerminalWithSearch {
+  getSearchAddon({
+    onDidChangeResults,
+  }: {
+    onDidChangeResults: ({
+      resultIndex,
+      resultCount,
+    }: {
+      resultIndex: number;
+      resultCount: number;
+    }) => void;
+  }): SearchAddon;
+}
 
 export const TerminalSearch = ({
   xTerm,
   show,
   onClose,
   onOpen,
+  focusTerm,
+  isSearchKeyboardEvent,
 }: {
   xTerm: TerminalWithSearch;
   show: boolean;
+  focusTerm(): void;
   onClose(): void;
   onOpen(): void;
+  isSearchKeyboardEvent: (e: KeyboardEvent) => boolean;
 }) => {
   const theme = useTheme();
   const searchInputRef = useRef<HTMLInputElement>();
   const [searchValue, setSearchValue] = useState('');
-  // const [showSearch, setShowSearch] = useState(false);
   const [searchResults, setSearchResults] = useState<{
     resultIndex: number;
     resultCount: number;
   }>({ resultIndex: 0, resultCount: 0 });
 
-  function search(value: string, prev?: boolean) {
-    if (!xTerm) {
-      return;
-    }
+  const searchAddon = xTerm?.getSearchAddon({
+    onDidChangeResults: setSearchResults,
+  });
```

Comment:
I figured this might be the cleanest way to do this. That way, the addon can always be loaded (like all the other addons), but if a component wants to fetch the `searchAddon` it has to provide a way to get the results. Seems like this best place to put it and makes most sense to me. Let me know what you think

Reply:
I'm not sure if I get it... We still get the entire `SearchAddon` in the component so we have access to `onDidChangeResults`. What does passing it through arguments give us? It seems that now we have two ways to provide the callback.

Now I've run the code and I think you might run into a problem where `searchAddon` was undefined, so you couldn't pass the callback. I think we should pass a terminal instance instead of the ref, so it is always defined:
```tsx
      <TerminalAddonsContainer>
        {refCtrl.current && props.terminalAddons?.(refCtrl.current)}
      </TerminalAddonsContainer>
```

Then we will be able to subscribe to the changes just once (`TerminalSearch`):
```ts
  useEffect(() => {
    searchAddon.onDidChangeResults(setSearchResults);
  }, [searchAddon]);
```